src/Application/Actions/Donations/Create.php

```diff
@@ -84,6 +92,14 @@ protected function action(): Response
             $donation = $this->donationRepository->buildFromApiRequest($donationData);
         }
 
+        if ($customerId !== $donation->getPspCustomerId()) {
```

Comment:
@NoelLH It seems redundant to me that we're passing the `customerId` both in the route parameters and in the request's body. Why not just in one place? In that case, I'm not sure we'd need this `if` block and instead I think we could rely on the middleware verifying that the incoming request to create a donation does indeed belong to the `customerId `specified when creating that donation.

Reply:
@alihejazi I think this redundancy is a sensible trade-off because the serialisation of the data object relies on `pspCustomerId` being available and persisted just like any other property. Removing it from the object definition means way more variation and complexity around this compared to other properties.

Contra-wise, removing it from the route means much more opaque (and therefore risky), messy and slow authentication middleware behaviour.